hibernate-core/src/test/java/org/hibernate/orm/test/any/annotations/EmbeddedAnyTest.java

```diff
@@ -78,6 +81,17 @@ public void testEmbeddedAny(SessionFactoryScope scope) {
 		);
 	}
 
+	@Test
+	@TestForIssue( jiraKey = "HHH-15323")
+	public void testEmbeddedTypeHql(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					final List<Foo> foos = session.createQuery( "from Foo f where type(f.fooEmbedded.bar) = Bar1" )
+							.list();
+				}
+		);
+	}
+
```

Comment:
What if the query is the other way around i.e. `Bar1 = type(f.fooEmbedded.bar)`? The handling in the `SemanticQueryBuilder` IMO needs to be made a bit more bulletproof to handle all cases.

Reply:
Thanks @beikov , test added